### ISSUE TITLE
Null Reference Exception at VectorTileFactory.cs (null VectorData) 

### ISSUE THREAD

**Issue by Kisamegr** [mapbox/mapbox-unity-sdk#1572]
* Unity version: 2019.3.5f1
  * Scripting Runtime Version: .Net 4.x
  * Scripting Backend: Mono
  * Api Compatibility Level: .Net Standard 2.0
* Mapbox SDK version: 2.1.1
* The platform you're building to: Windows PC
* A description of what you're trying to do: Trying to colourize a specific country on the map like the tutorial, but on runtime not on the style itself.
* Steps to recreate the bug if appropriate:
Create a new style using one of the default ones. Download the Natural Earth data and upload it to the style (as in this tutorial https://docs.mapbox.com/help/tutorials/style-single-country/). Then create/open an existing mapbox sample project, set your style id in the "Image" section, and the uploaded natural earth data set id in the "Map Layers" section. Then, add a "Landuse" feature and select the "ne_10m_admin_0_countries-5dl778" Data layer in the drop down. It should overlay another map on top of the original with a blue color. If you zoom out, you will notice exceptions happening in the editor and many of the blue tiles are corrupted/not generated properly.

It is nerve breaking that I can't find an easy/proper way to style a specific country in a world map.

The exception is thrown at VectorTileFactory.cs line 458, "tile.VectorData" is null

**Comment by brnkhy** [comment]
@Kisamegr before I try the new data layer stuff, can you tell me which line you're getting the null ref as well? And have you tried adding null check? Asking because that's the first thing I'll try as well.

**Comment by Kisamegr** [comment]
Yeah forgot to note the file and line, updated the original post. 
The first thing I did was indeed to add a null check, that removed the thrown exception. But it didn't fix anything else, the blue tiles are still not generated properly, I guess that's another bug.
